mods/default/functions.lua

```diff
@@ -459,6 +459,23 @@ local function leafdecay_on_timer(pos, def)
 
 	minetest.remove_node(pos)
 	minetest.check_for_falling(pos)
+
+	-- spawn a few particles for the removed node
+	local gravity = tonumber(minetest.settings:get("movement_gravity")) or 9.81
+	gravity = vector.new(0, -gravity, 0)
+	minetest.add_particlespawner({
+		amount = 8,
+		time = 0.001,
+		minpos = vector.subtract(pos, {x=0.5, y=0.5, z=0.5}),
+		maxpos = vector.add(pos, {x=0.5, y=0.5, z=0.5}),
+		minvel = vector.new(-0.5, 0, -0.5),
+		maxvel = vector.new(0.5, 1.0, 0.5),
```

Comment:
I think i might prefer min/max Y vel to be -0.5 and 0.5, so that the 'explosion' of leaves is consistent in all directions, instead of upwards biased.

Reply:
Hm, might be more realistic to have it be [-1, 0] or [0, 0] since leaves do not fall upwards when breaking apart.
How about that?

Reply:
Fine for me.